geowebcache/kml/src/main/java/org/geowebcache/service/kml/KMZHelper.java

```diff
@@ -71,11 +72,13 @@
                         tileLayer.getName(), gridSetId, 
                         linkGridLocs[i], mime, null, null, null);
                 
-                tile.setTileLayer(tileLayer);
-                
                 // Apply request filters
                 try {
+                    tile.setTileLayer(tileLayer);
```

Comment:
Minor nitpick, but this doesn't really need to be inside the try block, only need the `secDisp.checkSecurity(tile);` here.

Reply:
I can't recall why I changed this so I've put it back.